lib/util/errors.js

```diff
@@ -4,27 +4,47 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
+const REASONS = [
+  "LOCALDB_VERSION",
+  "NOT_INITIALIZED",
+  "INITIALIZED",
+  "MISSING_APP_KEY",
+  "WRONG_APP_KEY",
+  "CRYPTO",
+  "LOCKED",
+  "INVALID_ITEM",
+  "MISSING_ITEM",
+  "OFFLINE",
+  "AUTH",
+  "NETWORK",
+  "SYNC_LOCKED",
+  "GENERIC_ERROR"
+];
+
 /**
  * Errors specific to DataStore operations.
  *
  */
 class DataStoreError extends Error {
   /**
+   * @constructs
    * Creates a new DataStoreError with the given message and reason.
    *
-   * @param {string} message - The error message
-   * @param {Symbol} [reason=DataStoreError.GENERIC_ERROR] - The reason for the error
+   * @param {string} [reason=DataStoreError.GENERIC_ERROR] - The reason for the error
+   * @param {string} [message] - The error message
    */
-  constructor(message, reason) {
-    if ("symbol" === typeof message) {
-      reason = message;
-      message = "";
+  constructor(reason, message) {
+    if (-1 === REASONS.indexOf(reason)) {
```

Comment:
Not sure if we want to use [`Array#includes()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) versus `indexOf() === -1`. On that note, I'm not sure if we have any list of minimum browser support that we are planning, or if we should run this library through something like Babel to make it ES5 compliant (and minify it, etc).

Also, not sure if we should log any REASON misses. Like, if somebody typed `DataStoreError.GENERIC_ERRROR` instead of `DataStoreError.GENERIC_ERROR`.

Reply:
I can take a look at [`Array#includes()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/includes), but I've been hesitant to use APIs that I haven't had personal experience with on other platforms ... notably iOS WebKit (which is the target for `lockbox-ios`).

As far as minifying ... prove to me the size is an actual problem to deal with **here**, and not something that the consumers are better equiped for.